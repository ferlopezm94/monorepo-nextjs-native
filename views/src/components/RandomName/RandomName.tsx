import { apiClient } from '@acme/core';
import React, { useState, useEffect } from 'react';

export function RandomName(): JSX.Element {
  const [name, setName] = useState('');

  async function getNewRandomName() {
    const name = await apiClient.randomName.get();
    setName(name);
  }

  useEffect(() => {
    getNewRandomName();
  }, []);

  return (
    <>
      <p className='text-black font-bold mb-2'>Random names generator</p>
      <p className='text-black mb-2'>{name}</p>
      <button
        className='bg-blue-600 text-white font-bold rounded-full p-2 w-1/4'
        onClick={getNewRandomName}
      >
        Get new random name
      </button>
    </>
  );
}
