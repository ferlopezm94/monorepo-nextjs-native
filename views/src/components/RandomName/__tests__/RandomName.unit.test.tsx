import React from 'react';
import renderer from 'react-test-renderer';

import { RandomName } from './../RandomName';

describe('RandomName', () => {
  test('Should render', () => {
    const component = renderer.create(<RandomName />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
