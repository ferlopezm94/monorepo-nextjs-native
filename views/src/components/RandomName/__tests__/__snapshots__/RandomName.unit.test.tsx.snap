// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomName Should render 1`] = `
Array [
  <p
    className="text-black font-bold mb-2"
  >
    Random names generator
  </p>,
  <p
    className="text-black mb-2"
  >
    
  </p>,
  <button
    className="bg-blue-600 text-white font-bold rounded-full p-2 w-1/4"
    onClick={[Function]}
  >
    Get new random name
  </button>,
]
`;
